src/server/bitops_family.cc

```diff
@@ -577,12 +593,488 @@ void BitCount(CmdArgList args, ConnectionContext* cntx) {
   HandleOpValueResult(res, cntx);
 }
 
+enum class EncodingType { UINT, INT, NILL };
+
+struct CommonAttributes {
+  EncodingType type;
+  size_t encoding_bit_size;
+  size_t offset;
+};
+
+struct Nill {};
+using ResultType = std::variant<int64_t, Nill>;
+
+struct Overflow {
+  enum Policy { WRAP, SAT, FAIL };
+
+  static std::optional<Policy> PolicyFromStr(std::string_view policy) {
+    if (policy == "WRAP") {
+      return WRAP;
+    }
+    if (policy == "SAT") {
+      return SAT;
+    }
+    if (policy == "FAIL") {
+      return FAIL;
+    }
+    return {};
+  }
```

Comment:
Same here, `parser.Switch()`... (silent failure issue)

Reply:
As I said above, I did not use this partly because of the verbosity but also because the silent failure is not very helpful here.

Reply:
you can add methods with explicit failure

Reply:
How is it more verbose if its 2 lines vs 10 😆But your call after all

Reply:
welll that's also true :rofl: